### PR TITLE
Changed mock console server and client to use patch path 'spec/plan'

### PR DESCRIPTION
**Pull request by vbusch**
Signed-off-by: Vanessa <vbusch@redhat.com>

### Type of change

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: console/console-init/ui/src/Pages/AddressSpaceDetail/AddressList/AddressListPage.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ export const AddressListPage: React.FunctionComponent<IAddressListPageProps> = (
             Namespace: addressBeingEdited.namespace
           },
           jsonPatch:
-            '[{"op":"replace","path":"/Plan","value":"' +
+            '[{"op":"replace","path":"/spec/plan","value":"' +
```
Comment (k-wall) [inline console/console-init/ui/src/Pages/AddressSpaceDetail/AddressList/AddressListPage.tsx]: In light of the discussion, shouldn't this be /Spec/Plan

File: console/console-init/ui/src/Pages/AddressSpaceList/AddressSpaceListPage.tsx [hunk 2]
```diff
@@ -91,7 +91,7 @@ export const AddressSpaceListPage: React.FunctionComponent<AddressSpaceListPageP
             Namespace: addressSpaceBeingEdited.nameSpace
           },
           jsonPatch:
-            '[{"op":"replace","path":"/Plan","value":"' +
+            '[{"op":"replace","path":"/spec/plan","value":"' +
```
Comment (k-wall) [inline console/console-init/ui/src/Pages/AddressSpaceList/AddressSpaceListPage.tsx]: And here?

File: console/console-init/ui/src/Pages/AddressSpaceDetail/AddressList/AddressListPage.tsx (hunk 1 above)
Comment (vbusch) [reply console/console-init/ui/src/Pages/AddressSpaceDetail/AddressList/AddressListPage.tsx]: The path variable is simply passed through to the address client interface.  Therefore, it needs to be lowercase for the server to be able to evaluate it on the resource.
We can accept title case, if we want to be more consistent.  But we will need to add a translation in the resolver to make it lower case.  A simple string replace function would be error prone.
I think it is best to leave this matching the openshift objects.  But am willing to do the translation for consistency purposes.

**Comment by k-wall** [comment]
Mmmm.   I think the mistake here was the capitalisation on the fields in the GraphQL schema.  It really ought to match the capitalisation of `json` annotation on the Kubernetes type rather than the Go field name.    If we make the switch across the board I *think* this problem will just disappear.

The areas that will need to be changed:

- the graphql schemas
- the queries embedded in the console
- Connection and Link types in the console-server will all need `json` annotations.

File: console/console-init/ui/src/Pages/AddressSpaceDetail/AddressList/AddressListPage.tsx (hunk 1 above)
Comment (k-wall) [reply console/console-init/ui/src/Pages/AddressSpaceDetail/AddressList/AddressListPage.tsx]: My review comment is spurious.   I retract it.

**Comment by k-wall** [comment]
@vbusch i took the liberty of rebasing/force pushing your work against `console-server.`   I need to manually merge Queries into console/console-init/ui/src/queries.  My intent was to make no other changes.

**Comment by k-wall** [comment]
I think objectMeta needs to be renamed to metadata in the GraphQL layer.   The filtering/ordering is now working against the JSON field names, so to present consistency at the API layer, objectMeta => metadata.   Also it seems odd to leave the few GraphQL fields (Total etc) as upper case.  I think these should be changed too.

I'll work on PR addressing these issues and raise a PR against your branch.

File: console/console-init/ui/src/components/AddressSpaceList/AddressSpaceListFilter.tsx [hunk 3]
```diff
@@ -217,7 +217,7 @@ export const AddressSpaceListFilter: React.FunctionComponent<IAddressSpaceListFi
       } else {
         const obtainedList = response.data.addressSpaces.AddressSpaces.map(
           (link: any) => {
-            return link.objectMeta.Name;
+            return link.objectMeta.name;
```
Comment (k-wall) [inline console/console-init/ui/src/components/AddressSpaceList/AddressSpaceListFilter.tsx]: This needs to be `metadata`.

File: console/console-init/ui/src/components/AddressSpaceList/AddressSpaceListFilter.tsx [hunk 4]
```diff
@@ -263,7 +263,7 @@ export const AddressSpaceListFilter: React.FunctionComponent<IAddressSpaceListFi
       } else {
         const obtainedList = response.data.addressSpaces.AddressSpaces.map(
           (link: any) => {
-            return link.objectMeta.Namespace;
+            return link.objectMeta.namespace;
```
Comment (k-wall) [inline console/console-init/ui/src/components/AddressSpaceList/AddressSpaceListFilter.tsx]: This needs to be metadata.

**Pull request merged**